src/bin/nydusd/fusedev.rs

```diff
@@ -477,12 +508,13 @@ pub fn create_nydus_daemon(
         inflight_ops: Mutex::new(Vec::new()),
         result_receiver: Mutex::new(result_receiver),
         trigger: Arc::new(Mutex::new(trigger)),
-        threads: Mutex::new(Vec::new()),
+        state_machine_thread: Mutex::new(None),
+        fuse_service_threads: Mutex::new(Vec::new()),
```

Comment:
Why is reaping fuse threads important?
Only wait for state machine termination?

Reply:
> Why is reaping fuse threads important? Only wait for state machine termination?

Yep, seeking for waiting state machine and fuse service thread termination separately. Because daemon status is only updated after successful action. For example, `interrupt` action is executed, then wait fuse service workers done, finally the daemon status is changed from `RUNNING` to `READY`.